modules/mauth-authenticator-akka-http/src/main/scala/com/mdsol/mauth/akka/http/MAuthDirectives.scala

```diff
@@ -269,71 +218,66 @@ trait MAuthDirectives extends StrictLogging {
-    *         the request is rejected with a MissingHeaderRejection if the expected header is not present
-    */
-  private def extractAuthenticationHeaders(
-    request: HttpRequest,
-    authHeaderName: String,
-    timeHeaderName: String,
-    isReject: Boolean
-  ): Directive1[MauthHeaderValues] = {
-    if (request.getHeader(authHeaderName).isPresent) {
-      val mccAuthenticator = request.getHeader(authHeaderName).get().value()
-      if (request.getHeader(timeHeaderName).isPresent) {
-        val timeValue = request.getHeader(timeHeaderName).get().value
-        Try(timeValue.toLong).toOption match {
-          case Some(time: Long) =>
-            provide(new MauthHeaderValues(mccAuthenticator, time))
-          case None => {
-            val msg = s"$timeHeaderName header supplied with bad format: [$timeValue]"
-            logger.error(msg)
-            reject(MalformedHeaderRejection(headerName = timeHeaderName, errorMsg = msg, None))
-          }
-        }
-      } else {
-        reject(MissingHeaderRejection(timeHeaderName))
-      }
-    } else {
-      val msg = s"$authHeaderName header is not isPresent"
-      logger.error(msg)
-      if (isReject)
-        reject(MissingHeaderRejection(authHeaderName))
-      else
-        provide(new MauthHeaderValues("", 0L))
-    }
-  }
+  private def extractRequestHeader(request: HttpRequest, headerName: String): String =
+    request.headers
+      .find(_.name().compareToIgnoreCase(headerName) == 0)
+      .map(_.value())
+      .getOrElse("")
 
   /**
     * Extracts the authentication header value of the HTTP request header for latest version of Mauth
     *
-    * @param request HttpRequest
     * @param disableV1
     *        the flag to specify if Mauth V1 is disabled or not.
     *        If Mauth V1 is disabled, extracts the authentication header of MCC-Authentication only.
     *        Otherwise, extracts the authentication header of X-MWS-Authentication if MCC-Authentication header is not found.
     *
-    * @return Directive1[MauthHeaderValues] of Mauth authentication header values
+    * @return Directive1[MauthHeaderValues] of Mauth authentication header values for V1 or V2
     *         the request is rejected with a MissingHeaderRejection if the expected header is not present
     */
-  def extractLatestAuthenticationHeaders(request: HttpRequest, disableV1: Boolean): Directive1[MauthHeaderValues] = {
-    extractAuthenticationHeaders(request, `MCC-Authentication`.name, `MCC-Time`.name, false).flatMap { mauthHeader: MauthHeaderValues =>
-      {
-        if (!mauthHeader.authenticator.isEmpty) {
-          provide(mauthHeader)
+  def extractLatestAuthenticationHeaders(disableV1: Boolean): Directive1[MauthHeaderValues] = {
+    extractRequest.flatMap { request: HttpRequest =>
+      // Try to extract and verify V2 headers
+      val authenticator = extractRequestHeader(request, `MCC-Authentication`.name)
+      if (!authenticator.isEmpty) {
```

Comment:
* can use `nonEmpty`
* Should this name be something like `authenticationHeaderStr`?

Reply:
updated as suggested